Travis CI: test in PHP 7.3

simison opened the pull request:
### Add 7.3
Adds PHP v7.3 containers for CI testing. PHP v7.3 is scheduled for December 6th.

Note that 7.3 is aliassed to RC versions in Travis and it's currently running RC5:
> $ php --version
> PHP 7.3.0RC5 (cli) (built: Nov 12 2018 00:13:27) ( ZTS )

- Latest (final) RC6: https://secure.php.net/archive/2018.php#id2018-11-22-1
- v7.3 timeline: https://wiki.php.net/todo/php73
- v7.3 available in Travis, aliassed to RC versions: https://github.com/travis-ci/travis-ci/issues/9717#issuecomment-429564626 (

### Dropping 7.0?
With this, should we drop testing in 7.0 as security updates for it will end on Dec 5th 2018 (in a few days)?
http://php.net/supported-versions.php

<img width="1357" alt="image" src="https://user-images.githubusercontent.com/87168/49078364-7d61e400-f246-11e8-8512-b5b054210320.png">

simison committed:
Travis CI: test in PHP 7.3

simison commented:
This brings up two failed tests:
```
There were 2 failures:

1) WP_Test_Jetpack_PHP_Lint::test_php_lint
Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in ./json-endpoints/class.wpcom-json-api-site-settings-endpoint.php on line 823
Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in ./json-endpoints/class.wpcom-json-api-update-post-endpoint.php on line 762
Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in ./json-endpoints/class.wpcom-json-api-update-post-endpoint.php on line 771
Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in ./json-endpoints/class.wpcom-json-api-update-post-v1-1-endpoint.php on line 841
Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in ./json-endpoints/class.wpcom-json-api-update-post-v1-1-endpoint.php on line 850
Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in ./json-endpoints/class.wpcom-json-api-update-post-v1-2-endpoint.php on line 812
Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in ./json-endpoints/class.wpcom-json-api-update-post-v1-2-endpoint.php on line 821
Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? in ./modules/custom-css/csstidy/class.csstidy.php on line 858
Failed asserting that an array is empty.
/tmp/wordpress-master/src/wp-content/plugins/jetpack/tests/php/test_php-lint.php:24

2) WP_Test_Jetpack_Sync_SSO::test_sync_sso_is_two_step_required_filter_true
Failed asserting that null is true.
/tmp/wordpress-master/src/wp-content/plugins/jetpack/tests/php/sync/test_class.jetpack-sync-sso.php:16
```

jeherve commented:
I rebased after merging #11030, but some tests still fail, all because of issues with this:

```
PHP Notice:  compact(): Undefined variable: title in /tmp/wordpress-latest/src/wp-content/plugins/jetpack/tests/php/sync/test_class.jetpack-sync-plugins.php on line 198
```

jeherve committed:
Tests: define default values instead of relying on compact()

compact() now issues notices when a variable is not set.
secure.php.net/manual/en/function.compact.php#refsect1-function.compact-changelog

jeherve committed:
Travis: do not run PHP 7.3 tests on WP 4.9.9

WP 5.0 was made compatible with PHP 7.3 in this changeset:
https://core.trac.wordpress.org/changeset/44166
https://core.trac.wordpress.org/ticket/44416

Unit tests for WP Versions that come before that will always fail, because of the following errors:
compact(): Undefined variable: context
https://core.trac.wordpress.org/changeset/44166#file6

Once WP_BRANCH=previous becomes 5.0.0, we can update Travis to run PHP 7.3 tests for all our Matrix.

jeherve commented:
I've made some changes and this should now be good for a review. The tests now pass.

02ba8cc2b83a5ce5da03c47e39cb87d3e6e201ae is worth mentioning:

we will not run PHP 7.3 tests on WP 4.9.9.

WP 5.0 was made compatible with PHP 7.3 in this changeset:
https://core.trac.wordpress.org/changeset/44166
https://core.trac.wordpress.org/ticket/44416

Unit tests for WP Versions that come before that will always fail, because of the following errors:
`compact(): Undefined variable: context`
This was addressed in Core in WP 5.0 here:
https://core.trac.wordpress.org/changeset/44166#file6

Once WP_BRANCH=previous becomes 5.0, we can update Travis to run PHP 7.3 tests for all our Matrix.

kraftbj reviewed (approved):
I like it and it tests as expected.

jeherve commented on .travis.yml:
```diff
@@ -45,6 +45,15 @@ matrix:
     env: WP_TRAVISCI="yarn lint"
   - language: node_js
     env: WP_TRAVISCI="yarn test-dangerci-and-adminpage"
+  # We can't test PHP 7.3 with WP_BRANCH=previous just yet, as previous is currently 4.9.9.
```
Since WordPress 5.1 will be out soon, I wonder if it isn't best to wait and then remove all this in a couple of weeks.

simison replied:
If it's better to wait until Feb 21st for WP 5.1 before merging this?

I'm pretty indifferent about it... Suppose it isn't much work to change this in a separate PR in a week. :-)

How ever you prefer! :-)

jeherve replied:
You're right. I'll merge this now. The more tests the better, even if it's only for a couple of weeks.

jeherve replied:
done in #11398
